src/asm/symbol.c

```diff
@@ -538,7 +538,7 @@ struct Symbol *sym_AddLocalLabel(char const *name)
 		 * as long, and then that this was the entirety of the `Parent` part of `name`.
 		 */
 		if (labelScope[i] != '\0' || name[i] != '.')
-			error("Not currently in the scope of '%.*s'\n", parentLen, name);
+			error("Not currently in the scope of '%.*s'\n", (int)parentLen, name);
```

Comment:
I would add at least a debug assertion that `parentLen` isn't truncated when converted to `int`. Especially since:

> A negative precision is taken as if the precision were omitted.

...which would mean buffer overread. I *think* that a truncation is currently impossible due to the buffer size, but this won't hold if #650 is fixed.

Maybe also a `static_assert` on the buffer size being its current numerical value, so that changing it will remind us to watch that assumption?

Reply:
Does mean you have a label with more then 2.1GB of characters. Sounds unlikely, and then it would just display the wrong string (the full label instead of only the scope) does that really require so much extra effort?

Reply:
I'd at least add an `assert` to make the *potential* bug more obvious.